composer.json

```diff
@@ -13,10 +13,11 @@
     ],
     "require": {
         "php": ">=5.5.0",
```

Comment:
Change this to `^7.2 || ^8.0`.

Reply:
And update the travis build matrix.

Reply:
What's the latest practice for PHP versions? What if someone wants to use PHP 9 nightly versions?

Reply:
As the last practice @GrahamCampbell  wrote `"php": "^7.2|^8.0",`

Reply:
Yep, having no upper bound on PHP versions is not a good idea. It is not possible to unmark a release as not compatible, but it is possible to do another release and mark it as compatible. Moreover, composer has a way to ignore only the PHP version constraints of packages, or simulate installing as if it were PHP 8.0.0, so this is a non-issue (I am aware that Symfony and PHPUnit have taken a different view on this matter, however, I disagree in the strongest possible terms for the reasons I just explained, and take the same view as Laravel, Laminas, phpspec (and also Guzzle 7)). In fact, Laminas and phpspec are so strict as to not even allow PHP 8.1 yet.

Reply:
I understand the reasons why marking a lower bound makes sense, but an upper will make a package uninstallable and you have to turn of requirement checking entirely which means you could install packages that require a larger php version than yours.

From this perspective, letting the user to fail sounds like a better option.

Are there any good articles or discussions explaining both pros/cons?

Reply:
@sagikazarmark I agree with graham. There was a problem because I use this on a production, the upper limit should come with the ci tests.

Reply:
> I understand the reasons why marking a lower bound makes sense, but an upper will make a package uninstallable and you have to turn of requirement checking entirely which means you could install packages that require a larger php version than yours.

Not true. You can use the other feature I mentioned and tell composer to simulate a PHP version different to the runtime version. You don't have to ignore the PHP requirements, although this is certainly what people first jump to usually.

Reply:
Please mark this thread as resolved.